scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1393,344 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _chandrupatla(func, a, b, *, args=(), xatol=_xtol, xrtol=_rtol,
```

Comment:
Private for now, but documented and tested to the standards of a public function. The intent is to add this to `root_scalar` in a separate PR. Whether it should be public on its own is left to a future decision.